apisix/init.lua

```diff
@@ -43,15 +43,18 @@ if ngx.config.subsystem == "http" then
 end
 local load_balancer
 local local_conf
-local dns_resolver
 local lru_resolved_domain
 local ver_header    = "APISIX/" .. core.version.VERSION
 
 
-local function parse_args(args)
-    dns_resolver = args and args["dns_resolver"]
-    core.utils.set_resolver(dns_resolver)
-    core.log.info("dns resolver", core.json.delay_encode(dns_resolver, true))
+local function parse_args()
+    local options = {}
+    core.utils.init_dns_proxy(options)
+    if options.nameservers then
```

Comment:
It's so strange here, you defined an empty `options` at line 51, in the meanwhile you check its contents here after calling `core.utils.init_dns_proxy`, it doesn't do any initializations for `options`.

You should fill the `options` with the config in `config.yaml`, like `dns_resolver`.

Reply:
I am not recognize that , their is a config  in yaml, thanks for your remind it

Reply:
fixed.